stdlib/public/Darwin/Accelerate/vDSP_Arithmetic.swift

```diff
@@ -0,0 +1,1907 @@
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func add<T, U, V>(_ vectorA: T,
+                                    to vectorB: U,
+                                    result: inout V)
+        where
+        T: _ContiguousCollection,
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        T.Element == Float, U.Element == Float, V.Element == Float {
+            
+            let n = result.count
+            precondition(vectorA.count == n && vectorB.count == n)
+            
+            result.withUnsafeMutableBufferPointer { r in
+                vectorA.withUnsafeBufferPointer { a in
+                    vectorB.withUnsafeBufferPointer { b in
+                        vDSP_vadd(a.baseAddress!, 1,
+                                  b.baseAddress!, 1,
+                                  r.baseAddress!, 1,
+                                  vDSP_Length(n))
+                    }
+                }
+            }
+    }
+    
+    /// Populates `result` with the elementwise sum of `vectorA` and `vectorB`,
+    /// double-precision.
+    ///
+    /// - Parameter vectorA: the `a` in `c[i] = a[i] + b[i]`.
+    /// - Parameter vectorB: the `b` in `c[i] = a[i] + b[i]`.
+    /// - Parameter result: The `c` in `c[i] = a[i] + b[i]`.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func add<T, U, V>(_ vectorA: T,
+                                    to vectorB: U,
+                                    result: inout V)
+        where
+        T: _ContiguousCollection,
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        T.Element == Double, U.Element == Double, V.Element == Double {
+            
+            let n = result.count
+            precondition(vectorA.count == n && vectorB.count == n)
+            
+            result.withUnsafeMutableBufferPointer { r in
+                vectorA.withUnsafeBufferPointer { a in
+                    vectorB.withUnsafeBufferPointer { b in
+                        vDSP_vaddD(a.baseAddress!, 1,
+                                   b.baseAddress!, 1,
+                                   r.baseAddress!, 1,
+                                   vDSP_Length(n))
+                    }
+                }
+            }
+    }
+    
+    // MARK: c[i] = a[i] - b[i]                             vDSP_vsub
+    
+    /// Populates `result` with the elementwise difference of `vector` and `fromVector`,
+    /// single-precision.
+    ///
+    /// - Parameter vectorA: the `b` in `c[i] = a[i] - b[i]`.
+    /// - Parameter vectorB: the `a` in `c[i] = a[i] - b[i]`.
+    /// - Parameter result: The `c` in `c[i] = a[i] - b[i]`.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func subtract<T, U, V>(_ vectorA: T,
+                                         from vectorB: U,
```

Comment:
Might I suggest that `subtract(_:by:result:)` (restoring the first two arguments to their lexical order in `a - b`) would be a more usable overlay function. I think users are _extremely_ likely to get the arguments wrong in `subtract(_:from:result:)`.

In terms of Swift naming, addition and multiplication do not require labels as the arguments are interchangeable, so they can be `add(_:_:result:)` and `multiply(_:_:result:)`.